onnx/test/test_external_data.py

```diff
@@ -672,5 +674,28 @@ def test_check_model_absolute(self) -> None:
             checker.check_model(self.model_filename)
 
 
+class TestRuntimeQuantizationFilePathBug(TestLoadExternalDataBase):
+    def test_load_from_path_object(self) -> None:
+        model_path = Path(self.model_filename)
+        model = onnx.load_model(model_path)
+
+    @skip_if_no_onnxruntime
+    def test_file_path_is_correct_during_quantization(self) -> None:
+        from onnxruntime.quantization import quantize_dynamic, QuantType
+        
+        prev_dir = os.getcwd()
+        os.chdir(os.path.dirname(self.temp_dir))
+
+        try:
+            quantize_dynamic(
+                model_input=self.model_filename,
+                model_output=os.path.join(self.temp_dir, f"quantized-{str(uuid.uuid4())}.onnx"),
+                weight_type=QuantType.QInt8,
+                use_external_data_format=True,
+            )
+        finally:
+            os.chdir(prev_dir)
```

Comment:
Can this be done with a context manager ?

Reply:
There's no built-in one for this; should I add one or maybe there's one that I couldn't find?